boot/freeldr/freeldr/arch/i386/hwdisk.c

```diff
@@ -139,10 +139,24 @@ DiskRead(ULONG FileId, VOID* Buffer, ULONG N, ULONG* Count)
     BOOLEAN ret;
     ULONGLONG SectorOffset;
 
+    if (DiskReadBufferSize == 0)
```

Comment:
Can we check that in `DiskOpen` instead?

Reply:
Done.